src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_RATE = 1.5;
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE;
+
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity(size + 1);
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index");
+        }
+        ensureCapacity(size + 1);
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        /*  exception 'Foreach not applicable to type 'core.basesyntax.List<T>'
+        how can we solve this?
 
+        for (T element : list) {
+            add(element);
+        }
+        */
```

Comment:
delete comments

Reply:
These comments are my questions. It would be nice if you answered them

Reply:
You can use for loop as you implemented only

Reply:
PR is not the right place for questions. We have java chat for it. 
Only ready-for-review code have to be present in the PR.